src/admin/server.rs

```diff
@@ -158,6 +158,37 @@ impl AdminServiceImpl {
         Ok(())
     }
 
+    pub async fn start_unit_on_vm(&self, vm: &str, unit: String) -> anyhow::Result<()> {
+        let vmname = format!("givc-{}.service", vm);
+
+        /* Return error if the vm is not registered */
+        let endpoint = self.agent_endpoint(&vmname)?;
+        let client = SystemDClient::new(endpoint);
+
+        /* Check status of the unit */
+        match client.get_remote_status(unit.clone()).await {
```

Comment:
May be also use `unit: &str` , and `unit.to_owned()` or `unit.into()` instead of clone()

Reply:
Done.